cpp/src/rolling/grouped_rolling.cu

```diff
@@ -228,6 +228,9 @@ namespace {
 template <typename T, CUDF_ENABLE_IF(cuda::std::numeric_limits<T>::is_signed)>
 __device__ T add_safe(T const& value, T const& delta)
 {
+  if constexpr (cuda::std::numeric_limits<T>::has_infinity) {
```

Comment:
Should this condition be `is_floating_point`? The branch also calls isnan, which is not explicitly tied to the condition.

Reply:
This is a fair point. I've been torn about this myself.

One option is to check for `has_infinity` for `infinity`, and `has_quiet_nan()` for `nan`. Another would be to check `std::numeric_limits<T>::is_iec559 == true`.

I think your suggestion might be most readable.

Reply:
Changed now.